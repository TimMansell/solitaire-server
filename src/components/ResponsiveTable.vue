<template>
  <div class="responsive-table">
    <div class="responsive-table__overlay" v-if="swiper">
      <img class="responsive-table__swipe" :src="swipeIcon" />
    </div>
    <Table :headings="headings" :items="items" />
  </div>
</template>

<script>
import swipeIcon from '@/assets/swipe.svg';
import Table from '@/components/Table.vue';

export default {
  name: 'ResponsiveTable',
  components: {
    Table,
  },
  data() {
    return {
      swipeIcon,
      swiper: true,
    };
  },
  props: {
    headings: {
      type: Array,
      default: () => [],
    },
    items: {
      type: Array,
      default: () => [],
    },
  },
  mounted() {
    setTimeout(this.hideSwiper, 3000);
  },
  methods: {
    hideSwiper() {
      this.swiper = false;
    },
  },
};
</script>

<style scoped lang="scss">
.responsive-table {
  position: relative;
  max-width: calc(100vw - var(--vr));
  overflow-x: auto;
  margin-bottom: var(--mg-md);

  @media (min-width: $bp-sm) {
    max-width: initial;
  }

  &__overlay {
    position: absolute;
    width: 100%;
    height: 100%;
    background: var(--col-secondary-alt);

    @media (min-width: $bp-sm) {
      display: none;
    }
  }

  &__swipe {
    position: relative;
    width: 3rem;
    height: 3rem;
    transform: translate(0%, 50%);
    animation: slide-left-right 1.5s ease-in-out alternate infinite;
  }
}

@keyframes slide-left-right {
  0% {
    transform: translate(0%, 50%);
  }

  10% {
    transform: translate(0%, 50%);
  }

  90% {
    transform: translate(100%, 50%);
  }

  100% {
    transform: translate(100%, 50%);
  }
}
</style>
