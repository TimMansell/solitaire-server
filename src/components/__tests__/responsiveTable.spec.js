import { shallowMount } from '@vue/test-utils';
import ResponsiveTable from '@/components/ResponsiveTable.vue';

const propsData = {
  headings: ['Heading 1', 'Heading 2', 'Heading 3'],
  items: [
    [1, 2, 3],
    [4, 5, 6],
  ],
};

describe('ResponsiveTable.vue', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(ResponsiveTable, {
      propsData,
      computed: {
        showTableHelper: () => true,
      },
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('does not show table helper', () => {
    const wrapper = shallowMount(ResponsiveTable, {
      propsData,
      computed: {
        showTableHelper: () => false,
      },
    });

    expect(wrapper.find('[data-test="responsive-table-helper"]').exists()).toBe(
      false
    );
  });
});
