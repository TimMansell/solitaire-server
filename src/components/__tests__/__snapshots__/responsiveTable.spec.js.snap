// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResponsiveTable.vue matches snapshot 1`] = `
<div class="responsive-table">
  <div data-test="responsive-table-helper" class="responsive-table__overlay"><img src="" class="responsive-table__swipe"></div>
  <table-stub headings="Heading 1,Heading 2,Heading 3" items="1,2,3,4,5,6" placeholderrows="1"></table-stub>
</div>
`;
